app/policies/sample_policy.rb

```diff
@@ -2,6 +2,15 @@
 
 # Policy samples authorization
 class SamplePolicy < ApplicationPolicy
+  def destroy_attachment?
+    project = Project.find(record.project_id)
```

Comment:
Since we are authorizing the `@sample` you can just do `record.project` in the calls below. We don't need to do a query here

Reply:
Removed in 8c4be64cd61a4a7fae8b642efa2a2c6c75d0ea7f